src/GithubPlugin.fs

```diff
@@ -127,27 +136,42 @@ type GithubPlugin() =
         | _ ->
             [ defaultResult ]
 
-    let tryRunApiSearch =
-           Async.Catch
-        >> Async.RunSynchronously
-        >> function
-            | Choice1Of2 result -> presentApiSearchResult result
-            | Choice2Of2 exn -> presentApiSearchExn exn
+    let tryRunApiSearch (result: Async<ApiSearchResult>) =
+        async {
+            let! res = result |> Async.Catch
 
+            return
+                match res with
+                | Choice1Of2 result -> presentApiSearchResult result
+                | Choice2Of2 exn -> presentApiSearchExn exn
+        }
     member this.ProcessQuery terms =
-        match parseQuery terms with
-        | RunApiSearch fSearch -> tryRunApiSearch fSearch
-        | SuggestQuery suggestion -> presentSuggestion suggestion
-
-    interface IPlugin with
-        member this.Init (context:PluginInitContext) =
+        async {
+            let! res =
+                match parseQuery terms with
+                | RunApiSearch fSearch -> tryRunApiSearch fSearch
+                | SuggestQuery suggestion -> async { return presentSuggestion suggestion }
+
+            return
+                res
+                |> List.map
+                    (fun r ->
+                        Result(
+                            Title = r.title,
+                            SubTitle = r.subtitle,
+                            IcoPath = "icon.png",
+                            Action = fun x -> r.action x
+                        ))
+                |> List<Result>
+        }
```

Comment:
`ProcessQuery` is meant as an intermediate function here, used in testing, which is why it should return F#-friendly types (`SearchResult list` instead of `List<Flow.Launcher.Plugin.Result>`)

Moving the mapping to `List<Flow.Launcher.Plugin.Result>` out of `ProcessQuery` will get you most of the way there to getting the tests back. 

The remaining issue to resolve is that `ProcessQuery` is now wrapped in `Async`. To resolve this, you can add a helper function in the tests code that wraps `ProcessQuery` with `Async.RunSynchronously` and then replace all instances of `plugin.ProcessQuery` with it.

Reply:
I check the test framework document. They have a testAsync CE, so we can simply replace test with testAsync😉

Reply:
Yeah I know, I use it daily 😛 I just suggested the more straightforward change to keep things simple.